tools/shared.py

```diff
@@ -2837,20 +2837,37 @@ def make_jscall(sig, sig_order=0, named=True):
 
   @staticmethod
   def make_invoke(sig, named=True):
-    legal_sig = JS.legalize_sig(sig) # TODO: do this in extcall, jscall?
-    args = ','.join(['a' + str(i) for i in range(1, len(legal_sig))])
-    args = 'index' + (',' if args else '') + args
+    if sig == 'X' and Settings.WASM:
```

Comment:
Nit: What happens if sig == 'X' but not settings.wasm? It should never happen, but if it does I think this would get weird. Might be more explicit as
```
if sig == 'X':
  assert Settings.WASM
```

Reply:
Good idea, done.